runners/flink/src/test/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/source/unbounded/FlinkUnboundedSourceReaderTest.java

```diff
@@ -92,7 +92,7 @@ public void testSnapshotStateAndRestore() throws Exception {
    * This is a concurrency correctness test. It verifies that the main thread is always waken up by
    * the alarm runner executed in the executor thread.
    */
-  @Test(timeout = 30000L)
+  @Test(timeout = 60000L)
   public void testIsAvailableAlwaysWakenUp() throws Exception {
     final int numFuturesRequired = 1_000_000;
```

Comment:
this test adds 1 million futures, sounds a bad choice in terms of unit testing. Would reduce the load of the test help here?

Reply:
Decreased to 1000.
I'd like to leave a 60 second timeout just in case.